packages/patternfly-4/react-core/src/components/Pagination/Pagination.js

```diff
@@ -0,0 +1,159 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import styles from '@patternfly/patternfly/components/Pagination/pagination.css';
+import { css } from '@patternfly/react-styles';
+import { DropdownDirection } from '../Dropdown';
+import Navigation from './Navigation';
+import OptionsMenu from './OptionsMenu';
+
+const perPageOptions = [
+  {
+    title: '10',
+    value: 10
+  },
+  {
+    title: '20',
+    value: 20
+  },
+  {
+    title: '50',
+    value: 50
+  },
+  {
+    title: '100',
+    value: 100
+  }
+]
+
+export const PaginationVariant = {
+  top: 'top',
+  bottom: 'bottom'
+}
+
+const propTypes = {
+  /** Position where pagination is rendered. */
+  variant: PropTypes.oneOf(Object.values(PaginationVariant)),
+  /** Number of items per page. */
+  perPage: PropTypes.number,
+  /** Select from options to number of items per page. */
+  perPageOptions: PropTypes.arrayOf(PropTypes.shape({
+    title: PropTypes.node,
+    value: PropTypes.number
+  })),
+  /** Total number of items. */
+  itemCount: PropTypes.number.isRequired,
+  /** Current page number. */
+  page: PropTypes.number,
+  /** First index of items on current page. */
+  itemsStart: PropTypes.number,
```

Comment:
not required but don't see in defaultProps, same with itemsEnd, amountOfPages, widgetId, dropDirection

Reply:
`itemsStart`, `itemsEnd` and `amountOfPages` is deliberate so consumers doesn't have to pass anything and best default value is undefined (in deault props I'd just assign `undefined` to these values).

On other hand `widgetId` and `dropDirection` should be in default props. Realy nice catch!